File: day:1/two_sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
       let mut hash: HashMap<i32, i32> = HashMap::new();
        for i in 0..nums.len() {
           let diff = (target - nums[i]);
            match hash.get(& diff) {
              Some(value) => {
                  return vec![*value, i as i32];
              }
                None => {
                    hash.insert(nums[i], i as i32);
                }
            }
        }

        return vec![0,0]; 
        }
}

File: day:1/twoSum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var hash = [Int:Int]()
        for (index, element) in nums.enumerated() {
            let diff = target - nums[index]
            if let j = hash[diff] {
                return [index, j]
            }
            hash[nums[index]] = index
        }
        return []
    }
}
